server.js

```diff
@@ -11,15 +11,29 @@ app.use(bodyParser.json());
 
 app.post('/update', function(req, res) {
     pg.connect(process.env.DATABASE_URL, function (err, conn, done) {
+        // watch for any connect issues
+        if (err) console.log(err);
         conn.query(
             'UPDATE salesforce.Contact SET Phone = $1, MobilePhone = $1 WHERE LOWER(FirstName) = LOWER($2) AND LOWER(LastName) = LOWER($3) AND LOWER(Email) = LOWER($4)',
-            [req.body.phone, req.body.firstName, req.body.lastName, req.body.email],
+            [req.body.phone.trim(), req.body.firstName.trim(), req.body.lastName.trim(), req.body.email.trim()],
             function(err, result) {
-                done();
                 if (err != null || result.rowCount == 0) {
-                    res.status(400).json({error: 'The specified contact was not found.'});
+                  conn.query('INSERT INTO salesforce.Contact (Phone, FirstName, LastName, Email) VALUES ($1, $2, $3, $4)',
```

Comment:
Also need to set `MobilePhone` since that is our little hidden field that we use to verify the user didn't cheat by updating the `Phone` in Salesforce.